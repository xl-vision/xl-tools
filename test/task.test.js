const lintMd = require('../src/task/lintMd')
const lintScss = require('../src/task/lintScss')
const lintTs = require('../src/task/lintTs')

describe('test tasks', () => {
    it('test lintMd', () => {
        expect.assertions(1)
        return lintMd({
            cwd: __dirname,
            src: 'files/lintMd.md'
        }).catch(err => {
            expect(err.message.replace(/.*lintMd.md/g, '')).toMatchSnapshot()
        })
    })

    it('test lintScss', () => {
        expect.assertions(1)
        return lintScss({
            cwd: __dirname,
            src: 'files/lintScss.scss'
        }).catch(err => {
            expect(err.message).toMatchSnapshot()
        })
    })

    it('test lintTs', () => {
        expect.assertions(1)
        return lintTs({
            cwd: __dirname,
            src: 'files/lintTs.ts'
        }).catch(err => {
            expect(err.message.replace(/:.*lintTs.ts/g, '')).toMatchSnapshot()
        })
    })

})