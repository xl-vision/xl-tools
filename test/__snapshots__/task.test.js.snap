// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test tasks test lintMd 1`] = `
": 1: MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: \\"123\\"]
: 1: MD041/first-line-heading/first-line-h1 First line in file should be a top level heading [Context: \\"*123*\\"]"
`;

exports[`test tasks test lintScss 1`] = `"Failed with 5 errors"`;

exports[`test tasks test lintTs 1`] = `"Failed to lint [1, 18]: file should end with a newline."`;
